drivers/video/video_stm32_dcmi.c

```diff
@@ -276,8 +278,8 @@ static int video_stm32_dcmi_stream_stop(const struct device *dev)
 		return -EIO;
 	}
 
-	/* Release the buffer allocated in stream_start */
-	k_heap_free(&video_stm32_buffer_pool, data->buffer);
+	/* Release the video buffer allocated in stream_start */
+	k_fifo_put(&data->fifo_in, data->vbuf);
```

Comment:
Shouldn't we 'release' it after HAL_DCMI_stop? to be sure no frame event can happen in between?

Reply:
This just puts the video buffer back on the queue, so even if a frame event occurs before the stop, the pointer should still be valid, so it's actually slightly better now because before it was free'd memory. I don’t mind making this fix, but if it will require approvals again, I’d prefer to send it in another PR. Note that this is where the old buffer was released.

Reply:
Ok, would be good to have a following patch, This is equivalent to disowning the buffer, you should not use anymore after that and can not speculate on what is going to happen to it.

Reply:
https://github.com/zephyrproject-rtos/zephyr/pull/84670